### ISSUE TITLE
Hosts not showing up in graphical hosts panel

### ISSUE THREAD

**Issue by Toontje** [rsmudge/armitage#188]
I ran a auxiliary/scanner/portscan/tcp on one host i know exists.
Scan works fine, however the host doesn't show up in the graphical hosts panel (see image).
Any ideas?
This is OSX 10.10, latest Metasploit, latest Armitage.
Armitage launched with /Applications/Armitage.app/Contents/MacOS/JavaAppLauncher -E

![screen shot 2015-12-02 at 07 55 25](https://cloud.githubusercontent.com/assets/4884243/11525350/4fe534d8-98ca-11e5-80bc-81f49c7d3881.png)

**Comment by Yassdx** [comment]
are you found  the sulotion??

**Comment by shadycuz** [comment]
I too have this same issue on kali rolling. Armitage worked the first time I ran it but stopped. I tried purging and reinstalling. Tried hosts -d on msf with no luck. When importing host with armitage they are added to msf they just never show up in the GUI.

**Comment by rsmudge** [comment]
See FAQ Item #1: http://www.fastandeasyhacking.com/faq

metasploit sometimes creates two workspaces named default in its database. This wreaks havoc with Armitage's ability to query hosts as it's going to query one of those workspaces named default but it can't query both of them.

**Comment by shadycuz** [comment]
Yeah I found that FAQ in my trouble shooting research. My MSF only has the
default workspace but for what ever reason my MSF and armitage start in
different databases. I can switch MSF to the database that armitage uses
after armitage starts up, but still the targets don't appear. I'm going to
try and track down the connection information for that default db in MSF
and switch it to the one armitage uses and maybe that will work if they
both start connected to the same one. If all else fails I'll reinstall them
both.

Thanks for the reply.

**Comment by shadycuz** [comment]
Just another update... when I start armitage and check msf: db_status it says connected to MSF3

But I can't find any reference to that database any where. 

When I start msfconsole from terminal and check db_status it says connected to MSF. 

Ive check the database.yml in /opt/metasploit/apps/pro/ui/config  /user/share/metasploit-framework/ and $home/.msf4 and none of these mention the "MSF3" database? 

When I click preferences in armitage and look at the db_connect string it says "MSF"  ???? 

Let me know what you think please.

**Comment by arismelachroinos** [comment]
solution?

**Comment by rsmudge** [comment]
There's a command to list workspaces in Metasploit. Run it and check to see if there are multiple workspaces named default. If there are, delete them with the same command. I think it's workspace and workspace -d.

I spent some time troubleshooting an issue like this with someone and we figured out that, in their MSF instance, they had two default workspaces. This absolutely wreaks havoc on Armitage's ability to grab the right information out of the database for you.
